test/integration/consul-container/test/upgrade/l7_traffic_management/traffic_management_test.go

```diff
@@ -0,0 +1,160 @@
+package l7trafficmanagement
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/consul/api"
+	libassert "github.com/hashicorp/consul/test/integration/consul-container/libs/assert"
+	libcluster "github.com/hashicorp/consul/test/integration/consul-container/libs/cluster"
+	libservice "github.com/hashicorp/consul/test/integration/consul-container/libs/service"
+	"github.com/hashicorp/consul/test/integration/consul-container/libs/utils"
+	"github.com/stretchr/testify/require"
+)
+
+// TestTrafficManagement_Upgrade Summary
+// This test starts up 3 servers and 1 client in the same datacenter.
+//
+// Steps:
+//   - Create a single agent cluster.
+//   - Create one static-server and 2 subsets and 1 client and sidecar, then register them with Consul
+//   - Validate static-server and 2 subsets are and proxy admin endpoint is healthy
+//   - Validate static servers proxy listeners should be up and have right certs
+func TestTrafficManagement_SetupServerAndClientWithSubsets(t *testing.T) {
+	t.Parallel()
+
+	type testcase struct {
+		oldversion    string
+		targetVersion string
+	}
+	tcs := []testcase{
+		{
+			oldversion:    "1.14",
+			targetVersion: utils.TargetVersion,
+		},
+	}
+
+	run := func(t *testing.T, tc testcase) {
+		cluster := createCluster(t, tc.oldversion)
```

Comment:
This can be replaced with `topology.NewPeeringCluster`

https://github.com/hashicorp/consul/blob/1fbfb5905b85dd8e6b073a794e463853a7e30e3d/test/integration/consul-container/test/observability/access_logs_test.go#L48